modules/dreamview/backend/teleop/teleop.cc

```diff
@@ -182,37 +216,94 @@ void TeleopService::UpdateModem(const std::string &modem_id,
+      for (int i = 0; i < daemon_rpt->services_size(); i++) {
+          // look for voip_encoder or encoder0..1.2
+          // check 'voip_encoder' first because it contains 'encoder'
+          std::string service =  daemon_rpt->services(i);
+          if (service.find("voip_encoder") != std::string::npos) {
+              voipIsRunning = true;
+          }
+          else if (service.find("encoder") != std::string::npos) {
+              runningEncoders ++;
+          }
+      }
+
+      // all  video encoders are running.
+      videoIsRunning = runningEncoders == encoder_count;
+
+      // we may need to write commands to start/stop the video stream
+      bool sendStartVideo = false;
+      bool sendStopVideo = false;
+
+      // scope for the lock
+      {
+        boost::unique_lock<boost::shared_mutex> writer_lock(mutex_);
+        teleop_status_["video"] = videoIsRunning;
+        teleop_status_["audio"] = voipIsRunning;
+
+        if  (teleop_status_["video"]) {
+          if (teleop_status_["video_starting"]) {
+            // video has started
+            teleop_status_["video_starting"] = false;
+          }
+          else if (teleop_status_["video_stopping"]) {
+            // not stopped yet
+            sendStopVideo = true;
+          }
+        }
+        // video not running
+        else {
+          if (teleop_status_["video_starting"]) {
+            // not started yet
+            sendStartVideo = true;
+          }
+          else if (teleop_status_["video_stopping"]) {
+            // video is stopped
+            teleop_status_["video_stopping"] = false;
+          }
+        }
+      }
+      if (sendStartVideo || sendStopVideo) {
+        SendVideoStreamCmd(sendStartVideo);
       }
-    }
-    boost::unique_lock<boost::shared_mutex> writer_lock(mutex_);
-    teleop_status_["video"] = aVideoEncoderIsRunning;
-    teleop_status_["audio"] = voipIsRunning;
   }
 }
 
 void TeleopService::UpdateOperatorDaemonRpt(
     const std::shared_ptr<DaemonServiceRpt> &daemon_rpt) {
   {
-    bool voipIsRunning = false;
-    for (int i = 0; i < daemon_rpt->services_size(); i++) {
-      std::string service = daemon_rpt->services(i);
-      if (service.find("voip_encoder") >= 0) {
-        voipIsRunning = true;
-        break;
+      bool voipIsRunning = false;
+      for (int i = 0; i < daemon_rpt->services_size(); i++) {
+          std::string service =  daemon_rpt->services(i);
+          if (service.find("voip_encoder") != std::string::npos) {
+              voipIsRunning = true;
+              break;
+          }
       }
-    }
-    boost::unique_lock<boost::shared_mutex> writer_lock(mutex_);
-    teleop_status_["mic"] = voipIsRunning;
+      boost::unique_lock<boost::shared_mutex> writer_lock(mutex_);
+      teleop_status_["mic"] = voipIsRunning;
+  }
+}
+
+void TeleopService::SendVideoStreamCmd(bool start_stop) {
+  DaemonServiceCmd msg;
+  if (start_stop) {
+    msg.set_cmd("start");
+  }
+  else {
```

Comment:
Please fix the style issue with `scripts/clang-format.sh`.

Reply:
fixed [here](https://github.com/ApolloAuto/apollo/pull/10106/commits/e52de49fccb6d559e548bf24fda32d77c7930df3)